nexus/db-queries/src/db/datastore/vpc.rs

```diff
@@ -1599,6 +1600,12 @@ mod tests {
                         external_dns_servers: Vec::new(),
                     },
                 };
+                let zone_config = BlueprintZoneConfig {
+                    config,
+                    // XXX: NotInService retains the previous test behavior --
+                    // we may wish to change this to InService.
+                    zone_policy: BlueprintZonePolicy::NotInService,
```

Comment:
Two thoughts:

* I suspect changing this won't affect the test...
* ... but maybe it should?

When we're resolving sleds that need firewall rules, should we include sleds that have "not in service" services? I am a little worried we've added machinery here around a policy that we're not actually using, so are wandering a bit from where we'd go if we were using it. (This is not a complaint about this PR; I strongly agree with its restructuring choices.)

Reply:
We discussed this in more detail in #oxide-update today. The conclusion that we came to is to fail to compile on the variants changing, for now, to be isomorphic to the current state. In #5211 we have to do a DB migration anyway, and there we may do a greater rethink of how this works (have a call about this tomorrow).

Reply:
Changed this to `InService` though!